packages/medusa/src/strategies/__tests__/tax-calculation.js

```diff
@@ -1,124 +1,177 @@
+          name: "Name 1",
+          rate: 8.25,
+        },
+        {
+          item_id: "item_1",
+          name: "Name 2",
+          rate: 12.5,
+        },
+      ],
+      context: {
+        shipping_address: null,
+        customer: {
+          email: "test@testson.com",
+        },
+        shipping_methods: [],
+        region: {
+          gift_cards_taxable: false,
+        },
+        allocation_map: {
+          item_1: {
+            discount: {
+              amount: 10,
+              unit_amount: 5,
+            },
+            gift_card: {
+              amount: 10,
+              unit_amount: 5,
+            },
           },
         },
       },
     },
-  },
-  {
-    title: "calculates correctly with gift card",
-
-    /*
-     * Subtotal = 2 * 100 = 200
-     * Taxable amount = 200 - 10 = 180
-     * Taxline 1 = 180 * 0.0825 = 15
-     * Taxline 2 = 180 * 0.125 = 23
-     * Total tax = 38
-     */
-    expected: 40,
-    items: [
-      {
-        id: "item_1",
-        unit_price: 100,
-        quantity: 2,
-      },
-    ],
-    taxLines: [
-      {
-        item_id: "item_1",
-        name: "Name 1",
-        rate: 8.25,
-      },
-      {
-        item_id: "item_1",
-        name: "Name 2",
-        rate: 12.5,
-      },
-    ],
-    context: {
-      shipping_address: null,
-      customer: {
-        email: "test@testson.com",
-      },
-      region: {
-        gift_cards_taxable: true,
-      },
-      shipping_methods: [],
-      allocation_map: {
-        item_1: {
-          discount: {
-            amount: 10,
-            unit_amount: 5,
-          },
-          gift_card: {
-            amount: 10,
-            unit_amount: 5,
+  ],
+  [
+    "calculates correctly with gift card",
+    {
+      /*
+       * Subtotal = 2 * 100 = 200
+       * Taxable amount = 200 - 10 = 180
+       * Taxline 1 = 180 * 0.0825 = 15
+       * Taxline 2 = 180 * 0.125 = 23
+       * Total tax = 38
+       */
+      expected: 40,
+      items: [
+        {
+          id: "item_1",
+          unit_price: 100,
+          quantity: 2,
+        },
+      ],
+      taxLines: [
+        {
+          item_id: "item_1",
+          name: "Name 1",
+          rate: 8.25,
+        },
+        {
+          item_id: "item_1",
+          name: "Name 2",
+          rate: 12.5,
+        },
+      ],
+      context: {
+        shipping_address: null,
+        customer: {
+          email: "test@testson.com",
+        },
+        region: {
+          gift_cards_taxable: true,
+        },
+        shipping_methods: [],
+        allocation_map: {
+          item_1: {
+            discount: {
+              amount: 10,
+              unit_amount: 5,
+            },
+            gift_card: {
+              amount: 10,
+              unit_amount: 5,
+            },
           },
         },
       },
     },
-  },
+  ],
 ]
 
+const taxInclusiveShippingTests = [
+  [
+    "calculates correctly with tax inclusive shipping",
+    {
+      expected: 25,
```

Comment:
question: should the result be 30?
25% includes in the 100 = 25
10% not included in the 50 = 5
25 + 5